libs/server/Storage/Session/MainStore/MainStoreOps.cs

```diff
@@ -1190,5 +1193,299 @@ public GarnetStatus MemoryUsageForKey<TContext, TObjectContext>(ArgSlice key, ou
+
+            // Write "matches" section identifier
+            while (!RespWriteUtils.WriteBulkString(CmdStrings.matches, ref curr, end))
+                ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+
+            // Write matches array
+            while (!RespWriteUtils.WriteArrayLength(matches.Count, ref curr, end))
+                ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+
+            foreach (var match in matches)
+            {
+                while (!RespWriteUtils.WriteArrayLength(withMatchLen ? 3 : 2, ref curr, end))
+                    ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+
+                while (!RespWriteUtils.WriteArrayLength(2, ref curr, end))
+                    ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+
+                while (!RespWriteUtils.WriteInteger(match.Start1, ref curr, end))
+                    ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+
+                while (!RespWriteUtils.WriteInteger(match.Start1 + match.Length - 1, ref curr, end))
+                    ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+
+                while (!RespWriteUtils.WriteArrayLength(2, ref curr, end))
+                    ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+
+                while (!RespWriteUtils.WriteInteger(match.Start2, ref curr, end))
+                    ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+
+                while (!RespWriteUtils.WriteInteger(match.Start2 + match.Length - 1, ref curr, end))
+                    ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+
+                if (withMatchLen)
+                {
+                    while (!RespWriteUtils.WriteInteger(match.Length, ref curr, end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+                }
+            }
+
+            // Write "len" section identifier
+            while (!RespWriteUtils.WriteBulkString(CmdStrings.len, ref curr, end))
+                ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+
+            // Write LCS length
+            while (!RespWriteUtils.WriteInteger(lcsLength, ref curr, end))
+                ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+        }
+
+        private static byte[] ComputeLCS(ReadOnlySpan<byte> str1, ReadOnlySpan<byte> str2, int minMatchLen)
+        {
```

Comment:
Lots of code duplication here... Could probably use a helper method or two.

Reply:
Done